### PR TITLE
boards: add separate u54/u74 targets for `hifive_{unleashed,unmatched}` boards

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR adds support for U54 and U74 targets for `hifive_unleashed` and `hifive_unmatched`, respectively. The following changes are applied:
* hifive_unmatched: switched from `ram0` to `l2lim` in `zephyr, sram` in board DTS
* added a Kconfig option that informs whether the FPU state tracking is imprecise, which for FU540 and FU740 it is (FPU state is reported dirty where, theoretically, it's not)
* created the following new targets: `hifive_unleashed//e51`, `hifive_unleashed//u54`, `hifive_unmatched//s7`, `hifive_unmatched//u74`

### CODE REVIEW

File: soc/sifive/sifive_freedom/fu500/Kconfig [hunk 1]
```diff
@@ -18,3 +18,9 @@ config SOC_SERIES_SIFIVE_FREEDOM_FU500
```
Comment (nordicjm) [inline soc/sifive/sifive_freedom/fu500/Kconfig:18]: not part of your commit but drop this empty line

File: soc/sifive/sifive_freedom/fu500/Kconfig [hunk 2]
```diff
@@ -18,3 +18,9 @@ config SOC_SERIES_SIFIVE_FREEDOM_FU500
 
 	select 64BIT
 	select INCLUDE_RESET_VECTOR
+
+config SOC_SIFIVE_FREEDOM_FU540_U54
+	bool
+
```
Comment (nordicjm) [inline soc/sifive/sifive_freedom/fu500/Kconfig:24]:
```suggestion
```
and get rid of empty line

File: soc/sifive/sifive_freedom/fu500/Kconfig.soc [hunk 3]
```diff
@@ -5,12 +5,20 @@ config SOC_SERIES_SIFIVE_FREEDOM_FU500
...
+config SOC_SIFIVE_FREEDOM_FU540_E51
+	bool
+	select SOC_SIFIVE_FREEDOM_FU540
+
+config SOC_SIFIVE_FREEDOM_FU540_U54
+	bool
+	select SOC_SIFIVE_FREEDOM_FU540
+
+config SOC_SERIES
+	default "fu500" if SOC_SERIES_SIFIVE_FREEDOM_FU500
```
Comment (nordicjm) [inline soc/sifive/sifive_freedom/fu500/Kconfig.soc:21]: these need to match, so `SOC_SERIES_SIFIVE_FREEDOM_FU500` = `sifive_freedom_fu500`

File: soc/sifive/sifive_freedom/fu700/Kconfig [hunk 4]
```diff
@@ -17,3 +17,9 @@ config SOC_SERIES_SIFIVE_FREEDOM_FU700
 
 	select INCLUDE_RESET_VECTOR
 	select 64BIT
+
+config SOC_SIFIVE_FREEDOM_FU740_U74
+	bool
+
```
Comment (nordicjm) [inline soc/sifive/sifive_freedom/fu700/Kconfig:23]: as above

File: soc/sifive/sifive_freedom/fu500/Kconfig (hunk 1 above)
Comment (fkokosinski) [reply soc/sifive/sifive_freedom/fu500/Kconfig]: Should be fixed now

File: soc/sifive/sifive_freedom/fu500/Kconfig (hunk 2 above)
Comment (fkokosinski) [reply soc/sifive/sifive_freedom/fu500/Kconfig]: Done

File: soc/sifive/sifive_freedom/fu700/Kconfig (hunk 4 above)
Comment (fkokosinski) [reply soc/sifive/sifive_freedom/fu700/Kconfig]: Done

File: soc/sifive/sifive_freedom/fu500/Kconfig.soc (hunk 3 above)
Comment (fkokosinski) [reply soc/sifive/sifive_freedom/fu500/Kconfig.soc:21]: Not sure I follow - we're not changing this in this PR, just moving it a bit in the file. Plus, if we'd do the change from "fu500" to "sifive_freedom_fu500", then the whole path would have to change from `soc/sifive/sifive_freedom/fu500` to `soc/sifive/sifive_freedom/sifive_freedom_fu500`, which seems a bit redundant.

**Review by ycsin — DISMISSED** [review]

**Review by ycsin — DISMISSED** [review]

**Comment by fkokosinski** [comment]
Hey @kartben, could you please take a look at this PR? Thanks :)

**Review by kartben — CHANGES_REQUESTED** [review]
please update the documentation pages of the boards, they are essentially containing non-working build instructions now :)

```
❯  west build -b hifive_unleashed samples/hello_world -p auto
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: C:/Users/kartben/zephyrproject/zephyr/samples/hello_world
-- CMake version: 3.27.0-rc4
-- Found Python3: C:/Python312/python.exe (found suitable version "3.12.3", minimum required is "3.10") found components: Interpreter
-- Cache files will be written to: C:/Users/kartben/zephyrproject/zephyr/.cache
-- Zephyr version: 4.0.0-rc3 (C:/Users/kartben/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
CMake Error at C:/Users/kartben/zephyrproject/zephyr/cmake/modules/boards.cmake:304 (message):
  Board qualifiers `/fu540` for board `hifive_unleashed` not found.  Please
  specify a valid board target.

  Valid board targets for hifive_unleashed are:

  hifive_unleashed/fu540/e51

  hifive_unleashed/fu540/u54
```

**Comment by fkokosinski** [comment]
> please update the documentation pages of the boards, they are essentially containing non-working build instructions now :)
> 
> ```
> ❯  west build -b hifive_unleashed samples/hello_world -p auto
> -- west build: generating a build system
> Loading Zephyr default modules (Zephyr base).
> -- Application: C:/Users/kartben/zephyrproject/zephyr/samples/hello_world
> -- CMake version: 3.27.0-rc4
> -- Found Python3: C:/Python312/python.exe (found suitable version "3.12.3", minimum required is "3.10") found components: Interpreter
> -- Cache files will be written to: C:/Users/kartben/zephyrproject/zephyr/.cache
> -- Zephyr version: 4.0.0-rc3 (C:/Users/kartben/zephyrproject/zephyr)
> -- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
> CMake Error at C:/Users/kartben/zephyrproject/zephyr/cmake/modules/boards.cmake:304 (message):
>   Board qualifiers `/fu540` for board `hifive_unleashed` not found.  Please
>   specify a valid board target.
> 
>   Valid board targets for hifive_unleashed are:
> 
>   hifive_unleashed/fu540/e51
> 
>   hifive_unleashed/fu540/u54
> ```

Ooops, thanks for spotting that. Should be fixed now.

**Review by kartben — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Review by tgorochowik — APPROVED** [review]

**Pull request merged**